src/pages/ProductList/ProductList.js

```diff
@@ -1,54 +1,75 @@
 import React, { useState, useEffect } from 'react';
-import { Link, Outlet } from 'react-router-dom';
-import Nav from './../../components/Nav/Nav';
-import Category from './Category/Category';
+import { useLocation, useNavigate } from 'react-router-dom';
+import ProductListLayout from './ProductListLayout/ProductListLayout';
 import FilterOpen from './FilterOpen/FilterOpen';
-import BaseInfo from '../../components/BaseInfo/BaseInfo';
-import Footer from './../../components/Footer/Footer';
+import Category from './Category/Category';
+import CategoryList from './CategoryList/CategoryList';
 import './ProductList.scss';
+import API from '../../config/config';
 
 const ProductList = () => {
+  const { allProducts, category } = API;
+  const location = useLocation();
+  const navigate = useNavigate();
   const [productList, setProductList] = useState([]);
+  const [categoryInfo, setCategoryInfo] = useState([]);
   const [isFilterOpen, setIsFilterOpen] = useState(false);
 
   useEffect(() => {
-    fetch('http://localhost:3000/data/category_products.json')
+    fetch(`${allProducts}/${location.search}`)
+      .then(res => res.json())
+      .then(data => setProductList(data.result));
+  }, [location.search]);
+
+  useEffect(() => {
+    fetch(`${category}/${location.search}`)
```

Comment:
```suggestion
useEffect(() => {
    fetch(API.category`${location.search}`)
```

구조분해할당 하지않고, 이렇게 해보세요~

Reply:
API를 불러온 것이 훨씬 잘 보여 좋은것 같습니다! 위 말씀하신대로 수정하였습니다!